### DISCUSSION TITLE
How to suppress the modal component closing?

### DISCUSSION THREAD

**Discussion by mitya33** [tusen-ai/naive-ui#4856] [General]
It seems the <modal> component closes if you click the positive or negative actions buttons and there is no way I can see to suppress this, even if I add my own callback function to fire when these buttons are clicked.

The only way I've managed this is to remove the v-model aspect, but then the modal _never_ closes.

I don't want to use programmatic modals (i.e. the API).

Is it possible to suppress closure?

**Comment by jahnli** [comment]
```
  const dialog = $dialog.warning({
    content: '内容'
    onPositiveClick: async () => {
      try {
      } catch (error) {
        return false;
      } finally {
        dialog.loading = false;
      }
    }
```
`return false `  ?

**Reply by mitya33** [reply]
Not sure if by the question mark you're asking if this works or not sure. Thanks, I'll try this. It's not documented, so far as I can see, though.

**Reply by jahnli** [reply]
https://www.naiveui.com/zh-CN/os-theme/components/dialog#DialogOptions-Properties

`onPositiveClick`  `The default behavior is closing the confirm. Return false or resolve false or Promise rejected will prevent the default behavior.`


If you want to avoid turning off the default behavior of the modal box, I think you can do it by `return false`

**Reply by mitya33** [reply]
Aha! Somehow I missed that. Thanks.
